diff -Naur original/src/ipset.c patched/src/ipset.c
--- original/src/ipset.c	2015-07-31 03:59:07.000000000 +0800
+++ patched/src/ipset.c	2015-11-03 16:59:16.068677944 +0800
@@ -211,19 +211,22 @@
 
 int add_to_ipset(const char *setname, const struct all_addr *ipaddr, int flags, int remove)
 {
-  int af = AF_INET;
-
-#ifdef HAVE_IPV6
-  if (flags & F_IPV6)
-    {
-      af = AF_INET6;
-      /* old method only supports IPv4 */
-      if (old_kernel)
-	return -1;
-    }
-#endif
-  
-  return old_kernel ? old_add_to_ipset(setname, ipaddr, remove) : new_add_to_ipset(setname, ipaddr, af, remove);
+//  int af = AF_INET;
+//
+//#ifdef HAVE_IPV6
+//  if (flags & F_IPV6)
+//    {
+//      af = AF_INET6;
+//      /* old method only supports IPv4 */
+//      if (old_kernel)
+//	return -1;
+//    }
+//#endif
+//  
+//  return old_kernel ? old_add_to_ipset(setname, ipaddr, remove) : new_add_to_ipset(setname, ipaddr, af, remove);
+  char command [255];
+  sprintf(command, "/srv/railgun/network/hacks.sh %s %s", inet_ntoa(ipaddr->addr.addr4), setname);
+  return system(command);
 }
 
 #endif
